include/h2o.h

```diff
@@ -275,6 +275,28 @@ typedef struct st_h2o_protocol_callbacks_t {
     int (*foreach_request)(h2o_context_t *ctx, int (*cb)(h2o_req_t *req, void *cbdata), void *cbdata);
 } h2o_protocol_callbacks_t;
 
+enum h2o_status_handler_type {
+    H2O_STATUS_HANDLER_SIMPLE,
+    H2O_STATUS_HANDLER_PER_THREAD,
+};
+typedef h2o_iovec_t (*simple_status_handler_cb)(h2o_globalconf_t *gconf, h2o_mem_pool_t *pool);
+typedef struct h2o_status_handler {
+    enum h2o_status_handler_type type;
+    h2o_iovec_t name;
+    union {
+        struct {
+            simple_status_handler_cb status_cb;
+        } simple;
+        struct {
+            void *(*alloc_context_cb)(h2o_req_t *src_req);
+            void (*per_thread_cb)(void *priv, h2o_context_t *ctx);
+            h2o_iovec_t (*assemble_cb)(void *ctx);
+            void (*done_cb)(void *ctx);
+        } per_thread;
+    };
```

Comment:
How about having a single type that optionally defines `per_thread_cb`, instead of having two distinct types.

For example, we could define three callbacks: `init`, `per_thread`, `final`, with the first two being optional callbacks. `final` should either a) gather the information obtained by calls to `per_thread` (that updates the struct costructed by `init`), or b) simply return the information available in a global storage (in this case, `init` and `per_thread` can be set to NULL).